src/Linters/UseCanonicalDataProviders.hack

```diff
@@ -0,0 +1,307 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use namespace HH\Lib\{C, Dict, Math, Str, Vec};
+
+final class UseCanonicalDataProvidersLinter extends AutoFixingASTLinter {
+  const type TContext = ClassishDeclaration;
+  const type TNode = FunctionDeclarationHeader;
+  // dict<provider, vec<(test, provider again)>>
+  const type TProviders = dict<string, vec<(string, string)>>;
+
+  const string T_DATAPROVIDER = 'Facebook\\HackTest\\DataProvider';
```

Comment:
Can you add a comment with a reason for this not being a `::class` constant? I'm guessing that it's so that it doesn't full-depend on hacktest?

Reply:
> 
> ### Dependencies
> 
> Since I need to reflect on the DataProvider attribute, we now depend on HackTest being installed.
> We could also hardcode `const string DATA_PROVIDER = 'Facebook\HackTest\DataProvider'` in the class and `HH_IGNORE_ERROR` the `$reflection_method->getAttributeClass(static::DATA_PROVIDER)`.
> At runtime, this will be the same this, but it makes our code typecheck without having hacktest installed.
> We do need a unit test that asserts that the dataprovider attribute still exists, in case hacktest changes the name.

You are very much right. Let me add a comment about that.